File: core/src/main/java/y2k/joyreactor/SendMessageRequest.java
package y2k.joyreactor;

import rx.Observable;

/**
 * Created by y2k on 10/2/15.
 */
public class SendMessageRequest {

    private String username;

    public SendMessageRequest(String username) {
        this.username = username;
    }

    public Observable<Void> request(String message) {
        return ObservableUtils.create(() -> {
            new HttpClient()
                    .beginForm()
                    .put("username", username)
                    .put("text", message)
                    .putHeader("X-Requested-With", "XMLHttpRequest")
                    .putHeader("Referer", "http://joyreactor.cc/private/list")
                    .send("http://joyreactor.cc/private/create");
            return null;
        });
    }
}

File: core/src/main/kotlin/y2k/joyreactor/services/requests/SendMessageRequest.kt
package y2k.joyreactor.services.requests

import rx.Observable
import y2k.joyreactor.common.ajax
import y2k.joyreactor.common.ioUnitObservable
import y2k.joyreactor.http.HttpClient

/**
 * Created by y2k on 10/2/15.
 */
class SendMessageRequest(private val httpClient: HttpClient) {

    fun request(username: String, message: String): Observable<Unit> {
        return ioUnitObservable {
            httpClient
                .buildRequest()
                .addField("username", username)
                .addField("text", message)
                .ajax("http://joyreactor.cc/private/list")
                .post("http://joyreactor.cc/private/create")
        }
    }
}